pkgs/development/libraries/fbthrift/default.nix

```diff
@@ -38,7 +38,9 @@ stdenv.mkDerivation rec {
     flex
   ];
 
-  cmakeFlags = lib.optionals stdenv.isDarwin [
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=ON"
```

Comment:
Weirdly, on darwin this needs to be set to `OFF` to avoid running into an issue very similar to https://github.com/facebook/fbthrift/issues/280 (yes, 5 year old issue).

Turns out the `fbthrift` package has _always_  just been building static on macos and building the shared version is broken. The only thing that changed is that in this release it wants `BUILD_SHARED_LIBS` set explicitly (because of https://github.com/facebook/fbthrift/commit/8a08254ef1a27741b39939e5f7aa8b9acb7d791f ?)

Reply:
Alright I've made the flag conditional on darwin.